lib/kubeclient/common.rb

```diff
@@ -24,6 +24,10 @@ def initialize_client(
         password: nil,
         bearer_token: nil,
         bearer_token_file: nil
+      },
+      socket_options: {
```

Comment:
I agree @abonas, I was just trying to be consistent with the way `ssl_options` & `auth_options` are handled. I can clean them all up either here or in a different PR, whichever you fancy.